client/packages/programs/src/JsonForms/common/hooks/useJSONFormsCustomError.ts

```diff
@@ -16,24 +16,30 @@ export const useJSONFormsCustomError = (
       return;
     }
     const currentErrors = core?.errors ?? [];
-    if (
-      customError &&
-      !currentErrors.find(
-        it => it.schemaPath === path && it.keyword === keyword
-      )
-    ) {
-      dispatch(
-        Actions.updateErrors([
-          ...currentErrors,
-          {
-            instancePath: `/${path}`.replace(/\./g, '/'),
-            message: customError,
-            schemaPath: path,
-            keyword,
-            params: {},
-          },
-        ])
-      );
+    const existingIndex = currentErrors.findIndex(
+      it => it.schemaPath === path && it.keyword === keyword
+    );
+    if (customError) {
+      if (existingIndex === -1)
+        dispatch(
+          Actions.updateErrors([
+            ...currentErrors,
+            {
+              instancePath: `/${path}`.replace(/\./g, '/'),
+              message: customError,
+              schemaPath: path,
+              keyword,
+              params: {},
+            },
+          ])
+        );
+      else {
+        const existingAction = currentErrors[existingIndex];
+        if (existingAction && existingAction?.message !== customError) {
+          existingAction.message = customError;
+          dispatch(Actions.updateErrors([...currentErrors]));
+        }
+      }
```

Comment:
There was a problem in this implementation in that if the error message changed for a single element (e.g. goes from "Too small" to "Too big", this CustomError hook wouldn't get updated, since it's only checking for `path` and `keyword`.

So I've added an additional check to also check the messages are different and if so, replace the existing array element with the new message.